synapse/handlers/sliding_sync/room_lists.py

```diff
@@ -500,24 +500,28 @@ async def _compute_interested_rooms_new_tables(
                 # depending on the `required_state` requested (see below).
                 partial_state_rooms = await self.store.get_partial_rooms()
 
+                # Fetch any rooms that we have not already fetched from the database.
+                subscription_sliding_sync_rooms = (
+                    await self.store.get_sliding_sync_room_for_user_batch(
+                        user_id,
+                        sync_config.room_subscriptions.keys()
+                        - room_membership_for_user_map.keys(),
+                    )
+                )
+                room_membership_for_user_map.update(subscription_sliding_sync_rooms)
+
                 for (
                     room_id,
                     room_subscription,
                 ) in sync_config.room_subscriptions.items():
                     # Check if we have a membership for the room, but didn't pull it out
                     # above. This could be e.g. a leave that we don't pull out by
                     # default.
-                    current_room_entry = (
-                        await self.store.get_sliding_sync_room_for_user(
-                            user_id, room_id
-                        )
-                    )
+                    current_room_entry = room_membership_for_user_map.get(room_id)
```

Comment:
We expect this to be available and should blow-up when it doesn't to fix the logic.

```suggestion
                    current_room_entry = room_membership_for_user_map[room_id]
```

Reply:
This won't be the case if the client asks for e.g. a room that they've never been in. This matches the previous behaviour where `get_sliding_sync_room_for_user` returns None if the user doesn't have an entry in `sliding_sync_membership_snapshots` for that room

Reply:
Ahh, I see. Hence the `# TODO: Handle rooms the user isn't in.` below